analyses/filter-mutation-frequencies-tables/filter-mutation-frequencies-tables.Rmd

```diff
@@ -0,0 +1,324 @@
+---
+title: "Filter Mutation Frequencies Tables"
+output: 
+  html_notebook:
+    toc: TRUE
+    toc_float: TRUE
+    toc_depth: 4
+author: Eric Wafula for Pediatric OpenTargets
+date: 01/10/2021
+---
+
+Purpose: Remove Ensembl (ESNG) gene identifier in the mutation frequency tables, including SNV, CNV and fusion that are not in GENCODE v38 and Ensembl package 104.
+
+
+## Usage
+
+To run this from the command line, use:
+```
+Rscript -e "rmarkdown::render('filter-mutation-frequencies-tables.Rmd', clean = TRUE)"
+```
+_This assumes you are in the modules directory of the repository, OpenPedCan-analysis/analyses/filter-mutation-frequency-tables._
+
+## Setup
+
+Load libraries
+```{r}
+# R analysis packages
+suppressWarnings(
+  suppressPackageStartupMessages(library(rtracklayer))
+)
+suppressPackageStartupMessages(library(tidyverse))
+suppressPackageStartupMessages(library(data.table))
+
+# Magrittr pipe
+`%>%` <- dplyr::`%>%`
+```
+
+
+Set up directories. 
+```{r}
+# directories for input and output files
+root_dir <- rprojroot::find_root(rprojroot::has_dir(".git"))
+data_dir <- file.path(root_dir, "data")
+analyses_dir <- file.path(root_dir, "analyses")
+module_dir <- file.path(analyses_dir, "filter-mutation-frequencies-tables")
+snv_results_dir <- file.path(analyses_dir,  "snv-frequencies/results")
+cnv_results_dir <- file.path(analyses_dir,  "cnv-frequencies/results")
+fusion_results_dir <- file.path(analyses_dir,  "fusion-frequencies/results")
+results_dir <- file.path(module_dir, "results")
+
+# Create results folder if it doesn't exist
+if (!dir.exists(results_dir)) {
+  dir.create(results_dir)
+}
+```
+
+## Get Ensembl gene identifiers
+
+Load GenCode gtf file
+```{r}
+# read file
+gtf <- file.path(data_dir, "gencode.v38.primary_assembly.annotation.gtf.gz")
+gencode_gtf <- rtracklayer::import(con = gtf) %>% as.data.frame() %>% tibble()
+```
+
+Get Ensembl identifiers
+```{r}
+# filter for gene rows and get ids
+ensg_ids <- gencode_gtf %>% 
+  dplyr::filter(type == "gene") %>% 
+  dplyr::mutate(gene_id = stringr::str_extract(gene_id, "ENSG\\d+")) %>% 
+  pull(gene_id) %>% 
+  unique() 
+ensg_ids <- ensg_ids[!is.na(ensg_ids)]
+
```

Comment:
One additional thing: could you please also read in PMTL files and see whether the ENSGs that we filtered are in PMTL:
```
pmtl_df <- readr::read_tsv("../gene_match/input/PMTL_v1.1.tsv")
```